recipes/spy/0.0.3/conanfile.py

```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+
+class SpyConan(ConanFile):
+    name = "spy"
+    version = "0.0.3"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://jfalcou.github.io/spy/"
+    description = "C++ 17 for constexpr-proof detection and classification of informations about OS, compiler, etc..."
+    topics = ("c++17", "config", "metaprogramming")
+    no_copy_source = True
+
+    _source_subfolder = "source_subfolder"
+
```

Comment:
```suggestion
    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "17")
        compiler = self.settings.compiler
        comp_version = self.settings.compiler.version
        compilers = {"Visual Studio": "16", "gcc": "7.3", "clang": "6", "apple-clang": "10.0"}
        min_version = compilers.get(compiler)
        if not min_version:
            self.output.warn("{} recipe lacks information about the {} compiler standard version support.".format(self.name, comp_version))

        elif Version(comp_version) < min_version:
            raise ConanInvalidConfiguration("{} requires a compiler that supports"
                                            " at least C++17. {} {} is not" 
                                            " supported.".format(self.name, compiler, comp_version))
```

C++17 is not supported by all compilers

Reply:
There is already a compiler version check, was added while you were writing the suggestion. 🙂  We don't really have a unified check now, so I'd like to keep it simple [for a few weeks?] until we release at least some of the cppstd tools.

Reply:
This has been added at line 19->27